tests/unit/tf/horovod/test_horovod.py

```diff
@@ -0,0 +1,79 @@
+import importlib
+import os
+
+import tensorflow as tf
+
+import merlin.models.tf as mm
+from merlin.datasets.advertising.criteo.dataset import default_criteo_transform
+from merlin.io.dataset import Dataset
+from merlin.models.tf.distributed.backend import hvd
+from merlin.models.utils.example_utils import workflow_fit_transform
+from merlin.schema.tags import Tags
+
+
+def test_import():
+    from merlin.models.tf.distributed.backend import hvd
+
+    horovod_found = importlib.util.find_spec("horovod")
+
+    assert (horovod_found and hvd is not None) or (not horovod_found and hvd is None)
+
+
+def test_horovod_multigpu(criteo_data, tmpdir, batch_size=11, learning_rate=0.03):
```

Comment:
This test is not able to run inside the context of a regular testrun... is there no way to reference the test using a subprocess call and adding a parameter that allows you to ignore the subprocess command if you are already in that context? Or maybe put this subprocess command in the init file... and keep this as a function without the test_ in the front of the name so it is ignored by pytest.

Reply:
This may not work... but have we tried to run it in the same context as other tests... we run other subprocesses in many other tests (not mpi based but still for example notebooks). If it was tried and didnt work, all good, just want to make sure we tried.

Reply:
I tried running it with `subprocess` and wasn't able to find a way to make it work in subprocess. When tests are executed with `horovod -np 2 hvd_wrapper.sh python -m pytest tests/unit`, the tests pass and I can see from logs it's using both GPUs. These tests should work both on CPU (by _not_ using `horovodrun`) and on GPU (by using `horovodrun`).